### PR TITLE
frontend: Migrate BM_Matchbox from WithClusterConfig to Connect

### PR DESCRIPTION
**Pull request by kyoto**

### CODE REVIEW

File: installer/frontend/components/bm-matchbox.jsx [hunk 1]
```diff
@@ -3,18 +3,27 @@ import classNames from 'classnames';
...
   BM_MATCHBOX_HTTP,
   BM_MATCHBOX_RPC,
   BM_OS_TO_USE,
 } from '../cluster-config';
 
+const form = new Form('BM_MATCHBOX_ADDRESS', [
+  new Field(BM_MATCHBOX_RPC, {default: '', validator: validate.hostPort}),
+], {
+  validator: cc => !validate.hostPort(cc[BM_MATCHBOX_HTTP]) &&
+      !validate.nonEmpty(cc[BM_OS_TO_USE]) &&
```
Comment (kans) [inline installer/frontend/components/bm-matchbox.jsx]: This validator will be triggered when BM_MATCHBOX_RPC changes but it looks like BM_OS_TO_USE is set from a GET.  Do the timings work out here?  You may need to convert BM_MATCHBOX_HTTP as well.

**Comment by kyoto** [comment]
ok to test

**Comment by kyoto** [comment]
ok to test

File: installer/frontend/components/bm-matchbox.jsx (hunk 1 above)
Comment (kyoto) [reply installer/frontend/components/bm-matchbox.jsx]: I ended up moving this logic back to `BM_Matchbox.canNavigateForward()` to work around the problem.

**Pull request merged**